recipes/cargs/all/conanfile.py

```diff
@@ -0,0 +1,64 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class CwalkConan(ConanFile):
+    name = "cargs"
+    description = "A lightweight getopt replacement that works on Linux, " \
+                  "Windows and macOS. Command line argument parser library" \
+                  " for C/C++. Can be used to parse argv and argc parameters."
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    homepage = "https://likle.github.io/cargs/"
+    topics = ("cargs", "cross-platform", "windows", "macos", "osx", "linux",
+              "getopt", "getopt-long", "command-line-parser", "command-line",
+              "arguments", "argument-parser")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'fPIC': True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared and self.settings.os == "Windows"
```

Comment:
Hey @likle ,
since you're the author of this library, you should know that using `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` for creating a dll is considered a hack.
Using CMake, it is easy to export symbols properly, abstracting away differences between compilers (gcc/clang/msvc) and platforms (Windows/Linux/Macos).

Check the following resources:
- https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html for generating a header,
- https://docs.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-declspec-dllexport?view=vs-2019 for more background on dllimport/dllexport
- https://cmake.org/cmake/help/latest/prop_tgt/LANG_VISIBILITY_PRESET.html for also hiding the symbols on gcc/clang to have the same behavior on gcc/clang as on Windows

Reply:
:bookmark: this for myself!